src/wazuh_modules/agent_upgrade/manager/wm_agent_upgrade_commands.c

```diff
@@ -224,13 +242,177 @@ static void wm_agent_upgrade_start_upgrades(cJSON *json_response, const cJSON* t
             agent_task = (wm_agent_task *)node->data;
             node = wm_agent_upgrade_get_next_node(&index, node);
 
+            if (WM_UPGRADE_SUCCESS == wm_agent_upgrade_send_wpk_to_agent(agent_task)) {
+
+
+                // TODO: Send update command to agent and update task to UPDATING
+
+
+            } else {
+
+
+                // TODO: Update task to UPDATING/ERROR
 
-            // TODO: Send WPK to agent and update task to UPDATING/ERROR
 
+            }
 
             wm_agent_upgrade_remove_entry(atoi(agent_key));
         }
     } else {
         mtwarn(WM_AGENT_UPGRADE_LOGTAG, WM_UPGRADE_NO_AGENTS_TO_UPGRADE);
     }
 }
+
+static int wm_agent_upgrade_send_wpk_to_agent(const wm_agent_task *agent_task) {
```

Comment:
Can we split `wm_agent_upgrade_send_wpk_to_agent` into smaller methods? It´s both really hard to review and will be very complex to test